pkg/catalog/catalog.go

```diff
@@ -1711,6 +1717,99 @@ func (c *Catalog) PrepareExpiredCommits(ctx context.Context, repositoryID string
 	return c.Store.SaveGarbageCollectionCommits(ctx, repository, previousRunID)
 }
 
+func createParquetFile(entries []DBEntry) (string, error) {
+	// TODO: Modify using a run ID generator (https://github.com/treeverse/lakeFS/issues/4469)
+	filename := uuid.New().String()
+	fw, err := local.NewLocalFileWriter(filename)
+	if err != nil {
+		return "", err
+	}
+	defer fw.Close()
+	pw, err := writer.NewParquetWriter(fw, new(DBEntry), int64(len(entries)))
+	if err != nil {
+		return "", err
+	}
+	// Compression type
+	pw.CompressionType = parquet.CompressionCodec_GZIP
+	for _, entry := range entries {
+		if err = pw.Write(entry); err != nil {
+			return "", err
+		}
+	}
+	if err = pw.WriteStop(); err != nil {
+		return "", err
+	}
+	_, err = fw.Seek(0, 0)
+	if err != nil {
+		return "", err
+	}
+	return filename, nil
+}
+
+func markToString(mark *graveler.GCUncommittedMark) (*string, error) {
+	if mark == nil {
+		return nil, nil
+	}
+	var buf bytes.Buffer
+	encoder := base64.NewEncoder(base64.StdEncoding, &buf)
+	err := json.NewEncoder(encoder).Encode(mark)
+	if err != nil {
+		return nil, err
+	}
+	result := buf.String()
+	return &result, nil
+}
+
+func (c *Catalog) PrepareGCUncommitted(ctx context.Context, repositoryID, runID, contToken string) (*graveler.GarbageCollectionRunMetadata, *string, error) {
+	if err := validator.Validate([]validator.ValidateArg{
+		{Name: "repository", Value: repositoryID, Fn: graveler.ValidateRepositoryID},
+	}); err != nil {
+		return nil, nil, err
+	}
+	repository, err := c.getRepository(ctx, repositoryID)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	mark := graveler.GCUncommittedMark{
+		BranchID: "",
+		Key:      []byte(""),
+	}
+	if contToken != "" {
+		err = json.NewDecoder(base64.NewDecoder(base64.StdEncoding, strings.NewReader(contToken))).Decode(&mark)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
```

Comment:
I think it's more of the controller responsibility. The Catalog role is to pass the mark, the controller role is to pass it conveniently using HTTP. So I guess we can use the actual type and let the controller perform the conversions.